### ISSUE TITLE
`prefer-interface`: detect intersection types

### ISSUE THREAD

**Issue by OliverJAsh** [cartant/eslint-plugin-etc#26]
I expected an error for `B`:

```ts
interface A {
  x: number;
}
type B = A & {
  y: string;
};
```

This should be written as:

```ts
interface A {
  x: number;
}
interface B extends A {
  y: string;
}
```

WDYT?

**Comment by cartant** [comment]
Yeah, I started working on exactly that - in a local branch - late last year. I've written the tests and about half of the implementation.

**Comment by cartant** [comment]
The 1.2.0 release now as an `allowIntersection` option for this rule.
